overlord/snapshotstate/snapshotmgr.go

```diff
@@ -267,6 +267,22 @@ func prepareRestore(task *state.Task) (snapshot *snapshotSetup, oldCfg map[strin
 	return snapshot, oldCfg, reader, nil
 }
 
+// marshalSnapConfig encodes cfg to JSON and returns raw JSON message, unless
+// cfg is nil - in this case nil is returned.
+func marshalSnapConfig(cfg map[string]interface{}) (*json.RawMessage, error) {
```

Comment:
thanks for the helper

Reply:
maybe for symmetry this kind of code also merits to be turned into a helper?

```
	rawCfg, err := configGetSnapConfig(st, snapshot.Snap)
	if err != nil {
		return nil, nil, nil, fmt.Errorf("internal error: cannot obtain current snap config for snapshot restore: %v", err)
	}

	if rawCfg != nil {
		if err := json.Unmarshal(*rawCfg, &oldCfg); err != nil {
			return nil, nil, nil, fmt.Errorf("internal error: cannot decode current snap config: %v", err)
		}
	}
```

there is code like that in prepareSave and prepareRestore

Reply:
Good idea, thanks.